frontend/my-adopthive-app/src/AdopteeList/AdopteeList.jsx

```diff
@@ -1,54 +1,58 @@
-import { useState, useEffect } from 'react';
+import { useState, useEffect, useContext } from 'react';
 import './AdopteeList.css';
-import AdopteeSideBar from '../AdopteeSideBar/AdopteeSideBar';
+import AdopterSideBar from '../AdopterSideBar/AdopterSideBar';
 import Spinner from '../Loading/Loading';
 import useLoading from '../useLoading/useLoading';
+import { UserContext } from '../UserContext';
 
 const AdopteeList = () => {
   const [adoptees, setAdoptees] = useState([]);
   const [preferences, setPreferences] = useState({});
   const [error, setError] = useState('');
+  const { user } = useContext(UserContext);
   const [matchResult, setMatchResult] = useState(null);
   const [isSideBarOpen, setIsSideBarOpen] = useState(false);
+  const [favourites, setFavourites] = useState([]);
   const { isLoading, startLoading, stopLoading } = useLoading();
+  const UserId = user.id
 
   useEffect(() => {
     startLoading();
-    const fetchAdoptees = async () => {
-      try {
-        const response = await fetch('http://localhost:3004/adoptee');
-        if (response.ok) {
-          const data = await response.json();
-          setAdoptees(data);
-        } else {
-          setError('Failed to fetch adopters');
-        }
-      } catch (error) {
-        setError('Error: ' + error.message);
-      } finally {
-        stopLoading();
-      }
-    };
     fetchAdoptees();
   }, []);
 
+  const fetchAdoptees = async () => {
+    try {
+      const response = await fetch('http://localhost:3004/adoptee');
+      if (response.ok) {
+        const data = await response.json();
+        setAdoptees(data);
+      } else {
+        setError('Failed to fetch adopters');
+      }
+    } catch (error) {
+      setError('Error: ' + error.message);
+    } finally {
+      stopLoading();
+    }
+  };
+
   const handleRankChange = (adopteeId, rank) => {
     setPreferences({ ...preferences, [adopteeId]: rank });
   };
 
   const handleSubmit = async (e) => {
     e.preventDefault();
     const preferenceList = Object.keys(preferences).map(adopteeId => ({
-      adopterId: parseInt(adopteeId),
+      adopteeId: parseInt(adopteeId),
```

Comment:
how did you catch this bug?

Reply:
The file displays the list of adoptees and we are changing to integer. Therefore it has to be adopteeId not adopterId